python-sdk/nuscenes/scripts/export_2d_annotations_as_json.py

```diff
@@ -0,0 +1,137 @@
+# nuScenes dev-kit.
+# Code written by Sergi Adipraja Widjaja, 2019.
+# Licensed under the Creative Commons [see license.txt]
+
+"""
+Export 2D annotations (xmin, ymin,xmax, ymax) from re-projections of our annotated 3D bounding boxes to a .json file.
+"""
+
+from nuscenes.nuscenes import NuScenes
+from nuscenes.utils.geometry_utils import view_points
+from nuscenes.utils.geometry_utils import box_in_image
+
+import numpy as np
+import json
+import argparse
+import os
+
+from typing import List
+from pyquaternion.quaternion import Quaternion
+from collections import OrderedDict
+from tqdm import tqdm
+
+
+def generate_record(ann_rec: dict,
+                    x1: float,
+                    y1: float,
+                    x2: float,
+                    y2: float,
+                    sample_data_token: str,
+                    filename: str) -> OrderedDict:
+    """
+    Generate one 2D annotation record given various informations on top of the 2D bounding box coordinates.
+    :param ann_rec: Original 3d annotation record.
+    :param x1: Minimum value of the x coordinate.
+    :param y1: Minimum value of the y coordinate.
+    :param x2: Maximum value of the x coordinate.
+    :param y2: Maximum value of the y coordinate.
+    :param sample_data_token: Sample data tolk
+    :param filename:
+    :return: A sample 2D annotation record.
+    """
+    repro_rec = OrderedDict()
+    repro_rec['sample_data_token'] = sample_data_token
+
+    ignored_keys = ['rotation', 'sample_token', 'size', 'token', 'translation']
```

Comment:
I'd suggest doing the opposite. Explicitly list which keys you copy from `ann_rec`. That makes it easier to know what will be stored.